sos/sos_parent/src/test/java/se/inera/intyg/common/sos_parent/model/converter/certificate/question/QuestionDodsdatumTest.java

```diff
@@ -155,27 +160,31 @@ void shouldIncludeValidationMandatoryExpression() {
             assertEquals("$" + DODSDATUM_JSON_ID, certificateDataValidationMandatory.getExpression());
         }
 
-        @Test
-        void shouldIncludeValidationShowType() {
-            final var question = QuestionDodsdatum.toCertificate(null, 0, texts);
-            assertEquals(CertificateDataValidationType.SHOW_VALIDATION, question.getValidation()[1].getType());
-        }
+        @Nested
+        class IncludeValidationShowTest extends ValidationShowTest {
 
-        @Test
-        void shouldIncludeValidationShowQuestionId() {
-            final var question = QuestionDodsdatum.toCertificate(null, 0, texts);
-            final var certificateDataValidationShow = (CertificateDataValidationShow) question.getValidation()[1];
-            assertEquals(DODSDATUM_SAKERT_DELSVAR_ID, certificateDataValidationShow.getQuestionId());
-        }
+            @Override
+            protected String getQuestionId() {
+                return DODSDATUM_SAKERT_DELSVAR_ID;
+            }
 
-        @Test
-        void shouldIncludeValidationShowExpression() {
-            final var question = QuestionDodsdatum.toCertificate(null, 0, texts);
-            final var certificateDataValidationShow = (CertificateDataValidationShow) question.getValidation()[1];
-            assertEquals(
-                String.format("exists('%s') && '%s'", DODSDATUM_SAKERT_JSON_ID, DODSDATUM_SAKERT_JSON_ID),
-                certificateDataValidationShow.getExpression()
-            );
+            @Override
+            protected String getExpression() {
+                return multipleAndExpression(
```

Comment:
Should we really use the same implementation in the test? Historically we have built the actual expression as a string...

Reply:
Fixed!